### PR TITLE
Fix `_flatten` for `Identity`

### PR DESCRIPTION
**Pull request by albi3ro**
Just a quick bugfix. The `Identity._flatten` method doens't need to store the `n_wires` hyperparamter.

### CODE REVIEW

File: tests/ops/test_identity.py [hunk 1]
```diff
@@ -15,11 +15,36 @@
...
+    # pylint: disable=protected-access
+    def test_flatten_unflatten(self, wires):
+        """Test the flatten and unflatten methods of identity."""
+        op = Identity(wires)
+        data, metadata = op._flatten()
+        assert data == tuple()
+        assert metadata[0] == qml.wires.Wires(wires)
+        assert metadata[1] == tuple()
+
+        new_op = Identity(*op.flatten())
```
Comment (albi3ro) [inline tests/ops/test_identity.py:34]:
```suggestion
        new_op = Identity._unflatten(*op._flatten())
```

**Review by timmysilv — APPROVED** [review]

File: tests/ops/test_identity.py [hunk 2]
```diff
@@ -15,11 +15,36 @@
...
+import pennylane as qml
 from pennylane import Identity
 
 
 @pytest.mark.parametrize("wires", [[0], [0, 1], ["a", "b", "c"], [100, "xasd", 12]])
 class TestIdentity:
+
+    # pylint: disable=protected-access
+    def test_flatten_unflatten(self, wires):
+        """Test the flatten and unflatten methods of identity."""
```
Comment (BorjaRequena) [inline tests/ops/test_identity.py:27]: Would it make sense to copy/paste the couple of pytree-related tests from `test_parametric_ops.py` to `test_non_parametric_ops.py` so we cover them all too?

**Review by BorjaRequena — COMMENTED** [review]
Looks good! I think we could test the non-parametric ops (see comment below) to catch any other potential issues that may have slipped through?

File: tests/ops/test_identity.py (hunk 2 above)
Comment (albi3ro) [reply tests/ops/test_identity.py:27]: added Identity to the list of `NON_PARAMETRIZED_OPERATIONS` in `test_parametric_ops.py`.  Still kinda weird that we have a bunch of non-parametric ops tested in `test_parametric_ops.py`, but I don't feel like actually fixing that weirdness.

Comment (BorjaRequena) [reply tests/ops/test_identity.py:26]: 😂 alright

**Review by BorjaRequena — APPROVED** [review]
Easy fix!

File: tests/ops/test_identity.py (hunk 2 above)
Comment (timmysilv) [reply tests/ops/test_identity.py:26]: reminds me of this classic python thing:
```pycon
>>> qml.ops.qubit.parametric_ops_multi_qubit.Hadamard
<class 'pennylane.ops.qubit.non_parametric_ops.Hadamard'>
```
the multi-qubit, parametric Hadamard operator.

**Pull request merged**